EXOtrigger MET & HT prompt monitoring 92X

sudhaahuja opened the pull request:
Update with the METmonitoring and HTmonitoring codes for the offline DQM monitoring (EXO trigger paths added)

kpedro88 committed:
temporarily disable setNoiseFlagsQIE11 for AlCa

sudhaahuja committed:
added HTmonitoring, HT plots inside METmonitor, added in ExoClient

sudhaahuja committed:
edited METmonitor with HT plots

sudhaahuja committed:
METmonitor and HTmonitor updated

cmsbuild commented:
A new Pull Request was created by  @sudhaahuja (sudha) for master.

It involves the following packages:

Calibration/HcalAlCaRecoProducers
DQMOffline/Trigger


@ghellwig, @arunhep, @cerminar, @dmitrijus, @cmsbuild, @franzoni, @vanbesien, @davidlange6 can you please review it and eventually sign? Thanks.
@ghellwig, @battibass, @mtosi, @jhgoh, @calderona, @HuguesBrun, @tocheng, @mmusich, @trocino, @rociovilar this is something you requested to watch as well.
@davidlange6 you are the release manager for this.

cms-bot commands are listed <a href="http://cms-sw.github.io/cms-bot-cmssw-cmds.html">here</a>

arunhep commented:
@sudhaahuja i think it needs to be rebased.

arunhep commented:
-1

fwyzard commented:
please rebase

sudhaahuja committed:
updates after comment on PR

cmsbuild commented:
Pull request #18978 was updated. @ghellwig, @arunhep, @cerminar, @dmitrijus, @cmsbuild, @franzoni, @vanbesien, @davidlange6 can you please check and sign again.

mtosi commented:
i will check the list of paths at CERN
(w/ the phone is not that straightforward)

i've just had a worry/doubt
you are using only one jets collection,
but do we have case in which one make use of both the AK4 and AK8 collections ?
for instances, one used for the offline selections and the other for the HT computation ?
or maybe both for the offline selections ?

can you comment on this, please ?

thanks !!

rappoccio commented:
Hi, 

Also, I still do not see this file:

from DQMOffline.Trigger.htMonitoring_cfi import htMonitoring


It is not in the release, did you forget to commit it?

mtosi commented:
are you refering to htMonitoring ?

it is automatically created by the FillDescription

mtosi commented:
I think you need to rebase ....

mtosi commented:
I still do not see the following paths
what am I missing ?

HLT_PFMETTypeOne140_PFMHT140_IDTight_v1
HLT_PFMETNoMu130_PFMHTNoMu130_IDTight_v9
HLT_PFMETNoMu140_PFMHTNoMu140_IDTight_v9
HLT_MonoCentralPFJet80_PFMETNoMu130_PFMHTNoMu130_IDTight_v9
HLT_MonoCentralPFJet80_PFMETNoMu140_PFMHTNoMu140_IDTight_v9
HLT_PFHT890_v7
HLT_PFHT780_v7
HLT_PFHT680_v7
HLT_PFHT590_v7
HLT_PFHT510_v7
HLT_PFHT430_v7
HLT_PFHT370_v7
HLT_PFHT250_v7
HLT_PFHT180_v7

rappoccio commented:
@mtosi OK, so if the code gets correctly set, we can just clone it for the B2G setup? Thanks.

mtosi commented:
I would say so
you will probably need to update / adjust / overwrite the relevant
parameters
(above all the genericTriggerEventFlag settings)
  mia

On Thu, Jun 1, 2017 at 10:47 AM, rappoccio <notifications@github.com> wrote:

> @mtosi <https://github.com/mtosi> OK, so if the code gets correctly set,
> we can just clone it for the B2G setup? Thanks.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/cms-sw/cmssw/pull/18978#issuecomment-305430492>, or mute
> the thread
> <https://github.com/notifications/unsubscribe-auth/AEt587jJUeQKAw78EDEs1hpW3DxKbl15ks5r_nqSgaJpZM4NoKed>
> .
>

rappoccio commented:
@mtosi yes, we did that. OK, we're basically ready to go, just waiting for the rebase here and we can submit the B2G triggers. 

@alefisico we're just about done.

mtosi commented:
thanks !!!
  mia

On Thu, Jun 1, 2017 at 11:44 AM, rappoccio <notifications@github.com> wrote:

> @mtosi <https://github.com/mtosi> yes, we did that. OK, we're basically
> ready to go, just waiting for the rebase here and we can submit the B2G
> triggers.
>
> @alefisico <https://github.com/alefisico> we're just about done.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/cms-sw/cmssw/pull/18978#issuecomment-305444234>, or mute
> the thread
> <https://github.com/notifications/unsubscribe-auth/AEt581VY6vS2eDjghKeW1ivpCYaeZ687ks5r_of3gaJpZM4NoKed>
> .
>

alefisico commented:
@rappoccio great Sal, thanks

sudhaahuja committed:
updated hltpaths in HTmonitor & other changes

cmsbuild commented:
Pull request #18978 was updated. @ghellwig, @arunhep, @cerminar, @dmitrijus, @cmsbuild, @franzoni, @vanbesien, @davidlange6 can you please check and sign again.

sudhaahuja commented:
@mtosi - to answer your question - In EXO, only Alejandro's paths are using AK8 jets, so we can look into this with him once he requests the integration of his paths.

arunhep commented:
this PR needs a rebase. @sudhaahuja can you please act on it?

mtosi commented:
@sudhaahuja
can you please update this PR ?

alefisico commented:
sorry @sudhaahuja which implementation are you talking about? the substructure triggers have already a JIRA ticket with the implementation in the menu: https://its.cern.ch/jira/browse/CMSHLT-1262

cmsbuild commented:
Pull request #18978 was updated. @cmsbuild, @dmitrijus, @vanbesien, @davidlange6 can you please check and sign again.

mtosi commented:
@dmitrijus can we start the integration test, please ?

fwyzard commented:
tracked by #19142

fwyzard commented:
tracked by #19142

fwyzard commented:
please test

cmsbuild commented:
The tests are being triggered in jenkins.

cmsbuild commented:
-1

Tested at: f33d87e5c040123af9024bff53e2e55266c4f885

You can see the results of the tests here:
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-18978/20423/summary.html

I found follow errors while testing this PR

**Failed tests:  RelVals**

* **RelVals**:

When I ran the RelVals I found an error in the following worklfows:
5.1 step2
<pre>runTheMatrix-results/5.1_TTbar+TTbarFS+HARVESTFS/step2_TTbar+TTbarFS+HARVESTFS.log

</pre>4.22 step5
<pre>runTheMatrix-results/4.22_RunCosmics2011A+RunCosmics2011A+RECOCOSD+ALCACOSD+SKIMCOSD+HARVESTDC/step5_RunCosmics2011A+RunCosmics2011A+RECOCOSD+ALCACOSD+SKIMCOSD+HARVESTDC.log

</pre>135.4 step2
<pre>runTheMatrix-results/135.4_ZEE_13+ZEEFS_13+HARVESTUP15FS+MINIAODMCUP15FS/step2_ZEE_13+ZEEFS_13+HARVESTUP15FS+MINIAODMCUP15FS.log

</pre>8.0 step5
<pre>runTheMatrix-results/8.0_BeamHalo+BeamHalo+DIGICOS+RECOCOS+ALCABH+HARVESTCOS/step5_BeamHalo+BeamHalo+DIGICOS+RECOCOS+ALCABH+HARVESTCOS.log

</pre>140.53 step3
<pre>runTheMatrix-results/140.53_RunHI2011+RunHI2011+RECOHID11+HARVESTDHI/step3_RunHI2011+RunHI2011+RECOHID11+HARVESTDHI.log

</pre>1306.0 step4
<pre>runTheMatrix-results/1306.0_SingleMuPt1_UP15+SingleMuPt1_UP15+DIGIUP15+RECOUP15+HARVESTUP15/step4_SingleMuPt1_UP15+SingleMuPt1_UP15+DIGIUP15+RECOUP15+HARVESTUP15.log

</pre>1330.0 step3
<pre>runTheMatrix-results/1330.0_ZMM_13+ZMM_13+DIGIUP15+RECOUP15+HARVESTUP15/step3_ZMM_13+ZMM_13+DIGIUP15+RECOUP15+HARVESTUP15.log

</pre>9.0 step4
<pre>runTheMatrix-results/9.0_Higgs200ChargedTaus+Higgs200ChargedTaus+DIGI+RECO+HARVEST/step4_Higgs200ChargedTaus+Higgs200ChargedTaus+DIGI+RECO+HARVEST.log

</pre>4.53 step4
<pre>runTheMatrix-results/4.53_RunPhoton2012B+RunPhoton2012B+HLTD+RECODR1reHLT+HARVESTDR1reHLT/step4_RunPhoton2012B+RunPhoton2012B+HLTD+RECODR1reHLT+HARVESTDR1reHLT.log

</pre>25.0 step4
<pre>runTheMatrix-results/25.0_TTbar+TTbar+DIGI+RECOAlCaCalo+HARVEST+ALCATT/step4_TTbar+TTbar+DIGI+RECOAlCaCalo+HARVEST+ALCATT.log

</pre>1000.0 step4
<pre>runTheMatrix-results/1000.0_RunMinBias2011A+RunMinBias2011A+TIER0+SKIMD+HARVESTDfst2+ALCASPLIT/step4_RunMinBias2011A+RunMinBias2011A+TIER0+SKIMD+HARVESTDfst2+ALCASPLIT.log

</pre>1003.0 step3
<pre>runTheMatrix-results/1003.0_RunMinBias2012A+RunMinBias2012A+RECODDQM+HARVESTDDQM/step3_RunMinBias2012A+RunMinBias2012A+RECODDQM+HARVESTDDQM.log

</pre>10042.0 step5
<pre>runTheMatrix-results/10042.0_ZMM_13+ZMM_13TeV_TuneCUETP8M1_2017_GenSimFull+DigiFull_2017+RecoFull_2017+ALCAFull_2017+HARVESTFull_2017/step5_ZMM_13+ZMM_13TeV_TuneCUETP8M1_2017_GenSimFull+DigiFull_2017+RecoFull_2017+ALCAFull_2017+HARVESTFull_2017.log

</pre>136.731 step4
<pre>runTheMatrix-results/136.731_RunSinglePh2016B+RunSinglePh2016B+HLTDR2_2016+RECODR2_2016reHLT_skimSinglePh_HIPM+HARVESTDR2/step4_RunSinglePh2016B+RunSinglePh2016B+HLTDR2_2016+RECODR2_2016reHLT_skimSinglePh_HIPM+HARVESTDR2.log

</pre>50202.0 step4
<pre>runTheMatrix-results/50202.0_TTbar_13+TTbar_13+DIGIUP15_PU50+RECOUP15_PU50+HARVESTUP15_PU50/step4_TTbar_13+TTbar_13+DIGIUP15_PU50+RECOUP15_PU50+HARVESTUP15_PU50.log

</pre>10024.0 step5
<pre>runTheMatrix-results/10024.0_TTbar_13+TTbar_13TeV_TuneCUETP8M1_2017_GenSimFull+DigiFull_2017+RecoFull_2017+ALCAFull_2017+HARVESTFull_2017/step5_TTbar_13+TTbar_13TeV_TuneCUETP8M1_2017_GenSimFull+DigiFull_2017+RecoFull_2017+ALCAFull_2017+HARVESTFull_2017.log

</pre>25202.0 step4
<pre>runTheMatrix-results/25202.0_TTbar_13+TTbar_13+DIGIUP15_PU25+RECOUP15_PU25+HARVESTUP15_PU25/step4_TTbar_13+TTbar_13+DIGIUP15_PU25+RECOUP15_PU25+HARVESTUP15_PU25.log

</pre>20434.0 step4
<pre>runTheMatrix-results/20434.0_TTbar_14TeV+TTbar_14TeV_TuneCUETP8M1_2023D10_GenSimHLBeamSpotFull14+DigiFullTrigger_2023D10+RecoFullGlobal_2023D10+HARVESTFullGlobal_2023D10/step4_TTbar_14TeV+TTbar_14TeV_TuneCUETP8M1_2023D10_GenSimHLBeamSpotFull14+DigiFullTrigger_2023D10+RecoFullGlobal_2023D10+HARVESTFullGlobal_2023D10.log

</pre>10824.0 step5
<pre>runTheMatrix-results/10824.0_TTbar_13+TTbar_13TeV_TuneCUETP8M1_2018_GenSimFull+DigiFull_2018+RecoFull_2018+ALCAFull_2018+HARVESTFull_2018/step5_TTbar_13+TTbar_13TeV_TuneCUETP8M1_2018_GenSimFull+DigiFull_2018+RecoFull_2018+ALCAFull_2018+HARVESTFull_2018.log

</pre>23234.0 step4
<pre>runTheMatrix-results/23234.0_TTbar_14TeV+TTbar_14TeV_TuneCUETP8M1_2023D18_GenSimHLBeamSpotFull14+DigiFull_2023D18+RecoFullGlobal_2023D18+HARVESTFullGlobal_2023D18/step4_TTbar_14TeV+TTbar_14TeV_TuneCUETP8M1_2023D18_GenSimHLBeamSpotFull14+DigiFull_2023D18+RecoFullGlobal_2023D18+HARVESTFullGlobal_2023D18.log

</pre>27434.0 step4
<pre>runTheMatrix-results/27434.0_TTbar_14TeV+TTbar_14TeV_TuneCUETP8M1_2023D17_GenSimHLBeamSpotFull14+DigiFull_2023D17+RecoFullGlobal_2023D17+HARVESTFullGlobal_2023D17/step4_TTbar_14TeV+TTbar_14TeV_TuneCUETP8M1_2023D17_GenSimHLBeamSpotFull14+DigiFull_2023D17+RecoFullGlobal_2023D17+HARVESTFullGlobal_2023D17.log

</pre>

cmsbuild commented:
Comparison not run due to runTheMatrix errors (RelVals and Igprof tests were also skipped)

Teemperor committed:
Use std::min instead of TMath::Min in JetUtil.h

TMath::Min doesn't compile because there is no include for ROOT's
math header in this file. So instead we just use the STL min function.

mtosi commented:
what about 
HLT_PFHT890_v*
HLT_PFHT780_v*
HLT_PFHT680_v*
HLT_PFHT590_v*
HLT_PFHT510_v*
HLT_PFHT430_v*
HLT_PFHT370_v*
HLT_PFHT250_v*
HLT_PFHT180_v*
?!?!?!?

sudhaahuja committed:
small correction in HT/MET clients

cmsbuild commented:
Pull request #18978 was updated. @ghellwig, @arunhep, @cerminar, @dmitrijus, @cmsbuild, @franzoni, @vanbesien, @davidlange6 can you please check and sign again.

mtosi commented:
please, rebase

sudhaahuja committed:
Merge branch 'master' into EXOTrigger_promptMonitoring92X

cmsbuild commented:
Pull request #18978 was updated. @cmsbuild, @dmitrijus, @vanbesien, @davidlange6 can you please check and sign again.

sudhaahuja commented:
@mtosi - rebase done

fwyzard commented:
@cmsbuild please test

cmsbuild commented:
The tests are being triggered in jenkins.

sudhaahuja commented:
Also, @mtosi - the PFHT paths you are mentioning were already included in the branch sometime ago  -https://github.com/sudhaahuja/cmssw/blob/EXOTrigger_promptMonitoring92X/DQMOffline/Trigger/python/HTMonitor_cff.py#L118-#L170

dmitrijus commented:
how is this related to https://github.com/cms-sw/cmssw/pull/18971 ?

gomber commented:
@dmitrijus  it has nothing to do with #18971 , i think you got confused with the same branch name. But @sudhaahuja is only referring to her PR.

cmsbuild committed:
Merge pull request #19169 from Teemperor/AddTMathBaseIncludeToJetUtil.h

Use std::min instead of TMath::Min in JetUtil.h

cmsbuild commented:
-1

Tested at: fb31becd8190b5c3f7d6f9c9729d4b71a6333147


The following merge commits were also included on top of IB + this PR after doing git cms-merge-topic: 
https://github.com/cms-sw/cmssw/commit/fb31becd8190b5c3f7d6f9c9729d4b71a6333147
You can see more details here:
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-18978/20490/git-log-recent-commits
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-18978/20490/git-merge-result

You can see the results of the tests here:
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-18978/20490/summary.html

I found follow errors while testing this PR

**Failed tests:  RelVals**

* **RelVals**:

When I ran the RelVals I found an error in the following worklfows:
5.1 step2
<pre>runTheMatrix-results/5.1_TTbar+TTbarFS+HARVESTFS/step2_TTbar+TTbarFS+HARVESTFS.log

</pre>4.22 step5
<pre>runTheMatrix-results/4.22_RunCosmics2011A+RunCosmics2011A+RECOCOSD+ALCACOSD+SKIMCOSD+HARVESTDC/step5_RunCosmics2011A+RunCosmics2011A+RECOCOSD+ALCACOSD+SKIMCOSD+HARVESTDC.log

</pre>135.4 step2
<pre>runTheMatrix-results/135.4_ZEE_13+ZEEFS_13+HARVESTUP15FS+MINIAODMCUP15FS/step2_ZEE_13+ZEEFS_13+HARVESTUP15FS+MINIAODMCUP15FS.log

</pre>8.0 step5
<pre>runTheMatrix-results/8.0_BeamHalo+BeamHalo+DIGICOS+RECOCOS+ALCABH+HARVESTCOS/step5_BeamHalo+BeamHalo+DIGICOS+RECOCOS+ALCABH+HARVESTCOS.log

</pre>140.53 step3
<pre>runTheMatrix-results/140.53_RunHI2011+RunHI2011+RECOHID11+HARVESTDHI/step3_RunHI2011+RunHI2011+RECOHID11+HARVESTDHI.log

</pre>1306.0 step4
<pre>runTheMatrix-results/1306.0_SingleMuPt1_UP15+SingleMuPt1_UP15+DIGIUP15+RECOUP15+HARVESTUP15/step4_SingleMuPt1_UP15+SingleMuPt1_UP15+DIGIUP15+RECOUP15+HARVESTUP15.log

</pre>9.0 step4
<pre>runTheMatrix-results/9.0_Higgs200ChargedTaus+Higgs200ChargedTaus+DIGI+RECO+HARVEST/step4_Higgs200ChargedTaus+Higgs200ChargedTaus+DIGI+RECO+HARVEST.log

</pre>1330.0 step3
<pre>runTheMatrix-results/1330.0_ZMM_13+ZMM_13+DIGIUP15+RECOUP15+HARVESTUP15/step3_ZMM_13+ZMM_13+DIGIUP15+RECOUP15+HARVESTUP15.log

</pre>4.53 step4
<pre>runTheMatrix-results/4.53_RunPhoton2012B+RunPhoton2012B+HLTD+RECODR1reHLT+HARVESTDR1reHLT/step4_RunPhoton2012B+RunPhoton2012B+HLTD+RECODR1reHLT+HARVESTDR1reHLT.log

</pre>25.0 step4
<pre>runTheMatrix-results/25.0_TTbar+TTbar+DIGI+RECOAlCaCalo+HARVEST+ALCATT/step4_TTbar+TTbar+DIGI+RECOAlCaCalo+HARVEST+ALCATT.log

</pre>1000.0 step4
<pre>runTheMatrix-results/1000.0_RunMinBias2011A+RunMinBias2011A+TIER0+SKIMD+HARVESTDfst2+ALCASPLIT/step4_RunMinBias2011A+RunMinBias2011A+TIER0+SKIMD+HARVESTDfst2+ALCASPLIT.log

</pre>1003.0 step3
<pre>runTheMatrix-results/1003.0_RunMinBias2012A+RunMinBias2012A+RECODDQM+HARVESTDDQM/step3_RunMinBias2012A+RunMinBias2012A+RECODDQM+HARVESTDDQM.log

</pre>10042.0 step5
<pre>runTheMatrix-results/10042.0_ZMM_13+ZMM_13TeV_TuneCUETP8M1_2017_GenSimFull+DigiFull_2017+RecoFull_2017+ALCAFull_2017+HARVESTFull_2017/step5_ZMM_13+ZMM_13TeV_TuneCUETP8M1_2017_GenSimFull+DigiFull_2017+RecoFull_2017+ALCAFull_2017+HARVESTFull_2017.log

</pre>136.731 step4
<pre>runTheMatrix-results/136.731_RunSinglePh2016B+RunSinglePh2016B+HLTDR2_2016+RECODR2_2016reHLT_skimSinglePh_HIPM+HARVESTDR2/step4_RunSinglePh2016B+RunSinglePh2016B+HLTDR2_2016+RECODR2_2016reHLT_skimSinglePh_HIPM+HARVESTDR2.log

</pre>50202.0 step4
<pre>runTheMatrix-results/50202.0_TTbar_13+TTbar_13+DIGIUP15_PU50+RECOUP15_PU50+HARVESTUP15_PU50/step4_TTbar_13+TTbar_13+DIGIUP15_PU50+RECOUP15_PU50+HARVESTUP15_PU50.log

</pre>10024.0 step5
<pre>runTheMatrix-results/10024.0_TTbar_13+TTbar_13TeV_TuneCUETP8M1_2017_GenSimFull+DigiFull_2017+RecoFull_2017+ALCAFull_2017+HARVESTFull_2017/step5_TTbar_13+TTbar_13TeV_TuneCUETP8M1_2017_GenSimFull+DigiFull_2017+RecoFull_2017+ALCAFull_2017+HARVESTFull_2017.log

</pre>25202.0 step4
<pre>runTheMatrix-results/25202.0_TTbar_13+TTbar_13+DIGIUP15_PU25+RECOUP15_PU25+HARVESTUP15_PU25/step4_TTbar_13+TTbar_13+DIGIUP15_PU25+RECOUP15_PU25+HARVESTUP15_PU25.log

</pre>10824.0 step5
<pre>runTheMatrix-results/10824.0_TTbar_13+TTbar_13TeV_TuneCUETP8M1_2018_GenSimFull+DigiFull_2018+RecoFull_2018+ALCAFull_2018+HARVESTFull_2018/step5_TTbar_13+TTbar_13TeV_TuneCUETP8M1_2018_GenSimFull+DigiFull_2018+RecoFull_2018+ALCAFull_2018+HARVESTFull_2018.log

</pre>23234.0 step4
<pre>runTheMatrix-results/23234.0_TTbar_14TeV+TTbar_14TeV_TuneCUETP8M1_2023D18_GenSimHLBeamSpotFull14+DigiFullTrigger_2023D18+RecoFullGlobal_2023D18+HARVESTFullGlobal_2023D18/step4_TTbar_14TeV+TTbar_14TeV_TuneCUETP8M1_2023D18_GenSimHLBeamSpotFull14+DigiFullTrigger_2023D18+RecoFullGlobal_2023D18+HARVESTFullGlobal_2023D18.log

</pre>27434.0 step4
<pre>runTheMatrix-results/27434.0_TTbar_14TeV+TTbar_14TeV_TuneCUETP8M1_2023D17_GenSimHLBeamSpotFull14+DigiFullTrigger_2023D17+RecoFullGlobal_2023D17+HARVESTFullGlobal_2023D17/step4_TTbar_14TeV+TTbar_14TeV_TuneCUETP8M1_2023D17_GenSimHLBeamSpotFull14+DigiFullTrigger_2023D17+RecoFullGlobal_2023D17+HARVESTFullGlobal_2023D17.log

</pre>


The following merge commits were also included on top of IB + this PR after doing git cms-merge-topic: 
https://github.com/cms-sw/cmssw/commit/fb31becd8190b5c3f7d6f9c9729d4b71a6333147
You can see more details here:
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-18978/20490/git-log-recent-commits
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-18978/20490/git-merge-result

cmsbuild commented:
Comparison not run due to runTheMatrix errors (RelVals and Igprof tests were also skipped)

sudhaahuja committed:
Create ExoticaMonitoring_Client_cff.py

cmsbuild commented:
Pull request #18978 was updated. @cmsbuild, @dmitrijus, @vanbesien, @davidlange6 can you please check and sign again.

sudhaahuja commented:
There was a conflict in the file - ExoticaMonitoring_Client_cff.py earlier and resolving it directly on the PR resulted in few lines being deleted from the file, which led to the errors this time. Sorry for that.

Dr15Jones committed:
Added missing override

Dr15Jones committed:
Removed ELset.h

The typedef in ELset.h was only used in one file.

cmsbuild committed:
Merge pull request #19195 from Dr15Jones/removeELdestControl

Remove ELset.h

davidlange6 commented:
@sudhaahuja @gomber - if this and #18971 are not the same thing, please change their titles to be more descriptive of the changes. Thanks

dmitrijus commented:
+1

cmsbuild commented:
The tests are being triggered in jenkins.

cmsbuild commented:
This pull request is fully signed and it will be integrated in one of the next master IBs after it passes the integration tests. This pull request requires discussion in the ORP meeting before it's merged. @davidlange6, @smuzaffar

mtosi commented:
what about 

HLT_PFHT890_v*
HLT_PFHT780_v*
HLT_PFHT680_v*
HLT_PFHT590_v*
HLT_PFHT510_v*
HLT_PFHT430_v*
HLT_PFHT370_v*
HLT_PFHT250_v*
HLT_PFHT180_v*

sudhaahuja commented:
@mtosi - you can find those paths in the HT monitor here - https://github.com/sudhaahuja/cmssw/blob/EXOTrigger_promptMonitoring92X/DQMOffline/Trigger/python/HTMonitor_cff.py#L118-#L170

mtosi commented:
thanks, I've just re-checked in this PR, and I saw them ... sorry for not having spotted them before :(

can you please
1. fix the label of some paths
2. check the binning you use for the different HT threshold ?
thanks

cmsbuild commented:
-1

Tested at: 91ced54d2b5af6dc70278507de2b74ddbc454ae9

You can see the results of the tests here:
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-18978/20544/summary.html

I found follow errors while testing this PR

**Failed tests:  RelVals**

* **RelVals**:

When I ran the RelVals I found an error in the following worklfows:
1330.0 step3
<pre>runTheMatrix-results/1330.0_ZMM_13+ZMM_13+DIGIUP15+RECOUP15+HARVESTUP15/step3_ZMM_13+ZMM_13+DIGIUP15+RECOUP15+HARVESTUP15.log

</pre>

cmsbuild commented:
Comparison not run due to runTheMatrix errors (RelVals and Igprof tests were also skipped)

fwyzard commented on DQMOffline/Trigger/python/HTMonitor_cff.py:
```diff
@@ -0,0 +1,232 @@
+# HLT_MonoCentralPFJet80_PFMETNoMu130_PFMHTNoMu130_IDTight
+MonoCentralPFJet80_PFMETNoMu130_PFMHTNoMu130_HTmonitoring = hltHTmonitoring.clone()
+MonoCentralPFJet80_PFMETNoMu130_PFMHTNoMu130_HTmonitoring.FolderName = cms.string('HLT/HT/MonoCentralPFJet80_PFMETNoMu130/')
+MonoCentralPFJet80_PFMETNoMu130_PFMHTNoMu130_HTmonitoring.numGenericTriggerEventPSet.hltPaths = cms.vstring("MonoCentralPFJet80_PFMETNoMu130_PFMHTNoMu130_IDTight_v*")
```
please fix the path name to `HLT_MonoCentralPFJet80_PFMETNoMu130_PFMHTNoMu130_IDTight_v*`

fwyzard commented on DQMOffline/Trigger/python/HTMonitor_cff.py:
```diff
@@ -0,0 +1,232 @@
+# HLT_MonoCentralPFJet80_PFMETNoMu140_PFMHTNoMu140_IDTight
+MonoCentralPFJet80_PFMETNoMu140_PFMHTNoMu140_HTmonitoring = hltHTmonitoring.clone()
+MonoCentralPFJet80_PFMETNoMu140_PFMHTNoMu140_HTmonitoring.FolderName = cms.string('HLT/HT/MonoCentralPFJet80_PFMETNoMu140/')
+MonoCentralPFJet80_PFMETNoMu140_PFMHTNoMu140_HTmonitoring.numGenericTriggerEventPSet.hltPaths = cms.vstring("_v*")
```
please fix the path name to `HLT_MonoCentralPFJet80_PFMETNoMu140_PFMHTNoMu140_IDTight_v*`

fwyzard commented:
do you have the monitoring for
```
HLT_PFMETNoMu110_PFMHTNoMu110_IDTight_v*
HLT_PFMETNoMu130_PFMHTNoMu130_IDTight_v*
HLT_PFMETNoMu140_PFMHTNoMu140_IDTight_v*
```
?

rappoccio commented:
@sudhaahuja we're really running out of time here and we have B2G triggers that are also dependent upon this. What's the status? Do you need help with this?

sudhaahuja commented:
@rappoccio, others - sorry for the delay. I am having night shifts these days... I am working on it now & will commit the changes in a short while...

sudhaahuja committed:
Merged EXOTrigger_promptMonitoring92X from repository sudhaahuja

sudhaahuja commented:
@rappoccio, others - sorry for the delay. I am having night shifts these days... I am working on it now & will commit the changes in a short while...

rappoccio commented:
Thanks, if you need help, let us know. We can help out if needed.

sudhaahuja committed:
B2G monitor added from PR19214 & small changes HT/MET monitor

cmsbuild commented:
Pull request #18978 was updated. @smuzaffar, @Dr15Jones, @vazzolini, @silviodonato, @dmitrijus, @Martin-Grunewald, @kmaeshima, @cmsbuild, @fwyzard, @vanbesien, @davidlange6 can you please check and sign again.

sudhaahuja commented:
@mtosi @rappoccio @UBParker - the PR was updated with the changes/corrections suggested. I also added the B2G Monitor files provided by Ashley from this PR #19214 . @fwyzard - these triggers HLT_PFMETNoMu110_PFMHTNoMu110_IDTight_v*
HLT_PFMETNoMu130_PFMHTNoMu130_IDTight_v*
HLT_PFMETNoMu140_PFMHTNoMu140_IDTight_v*  , can be found inside MET monitor here - https://github.com/sudhaahuja/cmssw/blob/EXOTrigger_promptMonitoring92X/DQMOffline/Trigger/python/METMonitor_cff.py#L68-#L94

Martin-Grunewald commented:
-1
rebase properly

rappoccio commented:
@sudhaahuja thank you. Let us know if you need help with anything.

rappoccio commented:
@Martin-Grunewald I have this correctly rebased on a branch in my local repo. Is there a way to update THIS PR with that code, or should I make a new PR?

Martin-Grunewald commented:
Hi,

It looks to me that the list of commits includes commits from other PRs which should not be here, eg, https://github.com/cms-sw/cmssw/pull/18978/commits/815ec6fad1d3a87d5fe47e2cb0729caf63404ebd

That's why I think a rebase is needed (I assume you added these commits by committing to this branch from a later IB release area than your original, thus adding those spurious commits)

rappoccio commented:
@Martin-Grunewald Yes, that is what @sudhaahuja did, I suspect. However, she is now traveling. I have rebased these correctly on my fork. Should I make a new PR or is there a way to replace the commits in this one?

rappoccio commented:
I'm not sure how to proceed, so I opened the other PR: #19293. Should this one be closed?

Martin-Grunewald commented:
yes...
